test/bntests.pl

```diff
@@ -0,0 +1,162 @@
+#! /usr/bin/env perl
+# Copyright 2008-2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+# Run the tests specified in bntests.txt, as a check against OpenSSL.
+use strict;
+use warnings;
+use Math::BigInt;
+
+my $EXPECTED_FAILURES = 53;
+my $failures = 0;
+
+sub bn()
```

Comment:
With the `()`, you have declared that `bn` takes zero parameters, but when you use it, you violate that contract.

You can change that by either removing the parens:

``` perl
sub bn
```

or by declaring it takes one scalar argument:

``` perl
sub bn($)
```

This is the reason you were "forced" to use `&` in all calls.